### DISCUSSION TITLE
I2S and USB Audio

### DISCUSSION THREAD

**Discussion by yuri-evangelista** [earlephilhower/arduino-pico#1860] [Q&A]
Hi, wonderful work here! Thank you very much.

I have a question but not sure if anyone can help right now (maybe it will be solved by the next TinyUSB version).

I'm struggling with the problem of developing a USB microphone using a Pi Pico and 2 MEMS I2S mics.
At present I can:
* have the board recognized as an USB audio device  (but not I2S) using the mbed core and this library https://github.com/tierneytim/Pico-USB-audio
* have the I2S working (but no USB audio device) using the arduino-pico core
Is there a way to "merge" things in order to obtain a working USB microphone based on I2S mics?
Thank you very much!
Yuri

**Comment by frohro** [comment]
You might look at [Phil Schatzmann's fork of the Adafruit TinyUSB code.](https://www.pschatzmann.ch/home/2024/10/13/tinyusb-audio-on-an-rp2040-in-arduino/)  He has USB Audio working with Arduino-Pico.

**Comment by marcdraco1-gif** [comment]
The CMI108 will convert (probably a single) I2S mic direct to USB without needing to fiddle around with a Pico. I would think Phil's work would be more convenient though.
